### PR TITLE
Fixes the errors while running the project for Lumen to allow compatibility for Lumen as well

### PR DESCRIPTION
**Pull request by SachinBahukhandi**
Fixes: #50

### CODE REVIEW

**Comment by SachinBahukhandi** [comment]
Hey @viezel any plans on reviewing this, please?
It could be very much helpful for us like devs in using your package for lumen. Thanks!

File: src/Collectors/FrameworkCollector.php [hunk 1]
```diff
@@ -15,8 +15,14 @@ public function registerEventListeners(): void
         $startTime = defined('LARAVEL_START') ? LARAVEL_START : microtime(true);
         $this->addSegment('laravel boot', $startTime);
 
-        $this->app->booted(function () {
-            $this->endSegment('laravel boot');
-        });
+        if($this->app instanceof \Laravel\Lumen\Application){
```
Comment (viezel) [inline src/Collectors/FrameworkCollector.php:18]: You cannot refer to this->app due to you just deleted the proptery

File: src/Collectors/FrameworkCollector.php [hunk 2]
```diff
@@ -15,8 +15,14 @@ public function registerEventListeners(): void
...
+                $this->endSegment('laravel boot');
+        }
```
Comment (viezel) [inline src/Collectors/FrameworkCollector.php:20]: How did you test this logic? It seems incorrect as you have two endsection and incorrect type.

File: src/Collectors/EventsCollector.php [hunk 3]
```diff
@@ -8,14 +8,10 @@
 
 abstract class EventsCollector extends SegmentCollector
 {
-    /**
-     * @var \Illuminate\Foundation\Application
-     */
-    protected $app;
```
Comment (viezel) [inline src/Collectors/EventsCollector.php:14]: Why remove this?

**Review by viezel — CHANGES_REQUESTED** [review]
Cannot accept this

**Pull request closed without merging**